yarn-project/noir-contracts/src/contracts/multi_transfer_contract/src/main.nr

```diff
@@ -0,0 +1,132 @@
+    fn multiTransfer(
+        inputs: PrivateContextInputs,
+        asset: Field, // Asset to distribute
+        addresses: [Field; 12], // Addresses to distribute to
+        amounts: [Field; 12], // Amounts to distribute
+        owner: Field, // Owner of the asset
+        batch_transfer_selector: Field, // Function selector for transfer
+    ) -> distinct pub abi::PrivateCircuitPublicInputs {
+        let mut context = Context::new(inputs, abi::hash_args([
+            asset, 
+            addresses[0], 
+            addresses[1], 
+            addresses[2], 
+            addresses[3],
+            addresses[4],
+            addresses[5],
+            addresses[6], 
+            addresses[7], 
+            addresses[8], 
+            addresses[9],
+            addresses[10],
+            addresses[11],
+            amounts[0], 
+            amounts[1], 
+            amounts[2], 
+            amounts[3],
+            amounts[4],
+            amounts[5],
+            amounts[6], 
+            amounts[7], 
+            amounts[8], 
+            amounts[9],
+            amounts[10],
+            amounts[11],
+            owner,
+            batch_transfer_selector
+        ]));
+
+        // First batch transfer call
+        let return_values_1 = context.call_private_function(asset, batch_transfer_selector, [
+            owner, 
+            amounts[0],
+            amounts[1],
+            amounts[2],
+            addresses[0],
+            addresses[1],
+            addresses[2],
+            0 as Field,
+            0 as Field,
+        ]);
+        let result1 = return_values_1[0];
+        context.return_values.push(result1);
+
+        // Second batch transfer call
+        let return_values_2 = context.call_private_function(asset, batch_transfer_selector, [
+            owner, 
+            amounts[3],
+            amounts[4],
+            amounts[5],
+            addresses[3],
+            addresses[4],
+            addresses[5],
+            0 as Field,
+            0 as Field,
+        ]);
+        let result2 = return_values_2[0];
+        context.return_values.push(result2);
+
+        // Third batch transfer call
+        let return_values_3 = context.call_private_function(asset, batch_transfer_selector, [
+            owner, 
+            amounts[6],
+            amounts[7],
+            amounts[8],
+            addresses[6],
+            addresses[7],
+            addresses[8],
+            0 as Field,
+            0 as Field,
+        ]);
+        let result3 = return_values_3[0];
+        context.return_values.push(result3);
+
+        // Fourth batch transfer call
+        let return_values_4 = context.call_private_function(asset, batch_transfer_selector, [
+            owner, 
+            amounts[9],
+            amounts[10],
+            amounts[11],
+            addresses[9],
+            addresses[10],
+            addresses[11],
+            0 as Field,
```

Comment:
I guess pending commitments means that even in the next call if we fetch notes with 0 offset the same note wouldn't be return -> that's great!

So do we still need `spend_note_offset` and `spend_note_index` since both are always 0??

And if we still need them, shouldn't they be passed as fn parameters to `multiTransfer` - so offchain app devs compute what note offset to use and pass that instead of hard coding in an immutable contract.

Reply:
Good point, I think we should make them as parameters to the multiTransfer function. For the sake of this simple example, we don't need both of these parameters but the point is to show the developer an example of customising of spending notes.